python_modules/dagster/dagster/_serdes/ipc.py

```diff
@@ -188,13 +188,19 @@ def ipc_read_event_stream(file_path, timeout=30, ipc_process=None):
 
 
 def open_ipc_subprocess(parts, **kwargs):
-    """Sets new process group flags on Windows to support graceful termination."""
+    """Sets the correct flags to support graceful termination - the new subprocess will be in
+    its own process group so that CTRL-Cing the parent process doesn't immediately interrupt
+    the child process as well.
+    """
     check.list_param(parts, "parts", str)
 
     creationflags = 0
+    preexec_fn = None
     if sys.platform == "win32":
         creationflags = subprocess.CREATE_NEW_PROCESS_GROUP
-    return subprocess.Popen(parts, creationflags=creationflags, **kwargs)
+    else:
+        preexec_fn = os.setpgrp
```

Comment:
Does change what operating systems we can run on at all? Exclude any linux variants or anything? What about Mac?

Reply:
https://docs.python.org/3/library/os.html#os.setpgrp says "Unix, not Emscripten, not WASI." Mac is fine, that's what i'm testing on

Reply:
Ah WebAssembly stuff. Ok.

Reply:
That is worth a comment in this code path.